src/Components/Shared/src/WebEventData.cs

```diff
@@ -245,4 +258,22 @@ private static ChangeEventArgs DeserializeChangeEventArgs(string eventArgsJson)
             return changeArgs;
         }
     }
+
+    [JsonSerializable(typeof(WebEventDescriptor), GenerationMode = JsonSourceGenerationMode.MetadataAndSerialization)]
+    [JsonSerializable(typeof(WebEventDescriptor), GenerationMode = JsonSourceGenerationMode.MetadataAndSerialization)]
```

Comment:
nit: this is a duplicate of the LOC above. Probably an error case the generator should catch since the mode could be different and the intention would be unclear. We'd throw or use last-one-wins..